File: hw6/modules/newsapi.py
from modules.request import Request
from modules.secrets import NEWSAPI_API_KEY


class NewsAPI:
    BASE_URL = 'https://newsapi.org/v2'
    API_KEY = NEWSAPI_API_KEY

    @staticmethod
    def _build_url(resource):
        return f"{NewsAPI.BASE_URL}/{resource}"

    @staticmethod
    def get(url, params=None):
        if params is None:
            params = {}
        params['apiKey'] = NewsAPI.API_KEY
        return Request.get(url, params)

    @staticmethod
    def get_everything(q, page=1):
        url = NewsAPI._build_url('everything')
        return NewsAPI.get(url, {
            'q': q,
            'page': page
        })

File: hw8/server/src/newsapi.ts
import * as request from './request';

const config = {
    BASE_URL: 'https://newsapi.org/v2',
    API_KEY: 'fdf56dd5e25046118f12c4626ceac170'
};

const buildUrl = (resource: string): string => `${config.BASE_URL}/${resource}`;

const get = async (url: string, params: { [param: string]: any } = {}) => {
    params.apiKey = config.API_KEY;
    return await request.get(url, params);
};

const getEverything = async (q: string, page = 1) => {
    const url = buildUrl('everything');
    return await get(url, {
        'q': q,
        'page': page
    });
};
